webapp/packages/core-authentication/src/UserInfoResource.ts

```diff
@@ -126,15 +130,34 @@ export class UserInfoResource extends CachedDataResource<UserInfo | null, void,
     });
 
     if (authInfo.userTokens && authInfo.authStatus === AuthStatus.Success) {
-      this.resetIncludes();
-      this.setData(await this.loader());
-      this.sessionResource.markOutdated();
+      await this.syncData();
     }
 
     return authInfo as AuthInfo;
   }
 
-  finishFederatedAuthentication(authId: string, linkUser?: boolean): ITask<UserInfo | null> {
+  async asyncAuthLogin(provider: string, { configurationId, linkUser, forceSessionsLogout }: IAsyncLoginOptions): Promise<AsyncAuthInfo> {
+    const { result } = await this.graphQLService.sdk.asyncAuthLogin({
+      provider,
+      configuration: configurationId,
+      linkUser,
+      forceSessionsLogout,
+    });
+
+    return result;
+  }
+
+  async getAuthTaskResult(taskId: string): Promise<AuthTokenFragment[]> {
+    const { result } = await this.graphQLService.sdk.getAuthTaskResult({ taskId });
+
+    if (result.userTokens) {
+      await this.syncData();
+    }
```

Comment:
why and what happening here?

Reply:
This method could be removed as we dont rely on tokens from the call